lib/factory_bot/strategy/stub.rb

```diff
@@ -43,13 +43,17 @@ def result(evaluation)
 
       private
 
-      def next_id
-        @@next_id += 1
+      def next_id(result_instance)
+        if result_instance.respond_to?(:column_for_attribute) && result_instance.column_for_attribute("id").sql_type == "uuid"
```

Comment:
I don't think `sql_type` is part of the public rails api, but I'm not sure of another way to get this.

We should us `result_instance.class.primary_key` instead of `"id"` for classes with a custom primary key.

Let's also extract this conditional into a private helper method like `uuid_primary_key?`